saleor/checkout/complete_checkout.py

```diff
@@ -573,8 +667,24 @@ def complete_checkout(
     :raises ValidationError
     """
     checkout = checkout_info.checkout
-    channel_slug = checkout_info.channel.slug
-    payment = checkout.get_last_active_payment()
+
+    # For individual payments, checkoutComplete will process the payment,
+    # otherwise a specific paymentID has to be provided
+    incomplete_payments = [
+        p for p in get_active_payments(checkout) if not p.is_authorized
+    ]
+
+    if payment is None:
+        if len(incomplete_payments) > 1:
```

Comment:
What about case:
As a customer, I've create a payment object, then changed something in checkoutLines, change addres etc, then again, I am moving to the payment step, which will create a second incomplete payment.  This condition will not allow me to finish a checkout. Am I right?

Reply:
Creating a subsequent payment with partial=False (default) will cancel/refund/mark as inactive all previous payments.

Otherwise it's already an implementation that is aware of partial payments and will provide the necessary paymentID